### ISSUE TITLE
Trainer stuck in infinite validation loop 

### ISSUE THREAD

**Issue by jaswanthbjk** [Lightning-AI/lightning#17813]
### Bug description

I am training a model with ---min_epochs flag and an early stopping callback. 

After the early stopping is triggered but the trainer should continue due to the min_epochs flag. The trainer is stuck only performing validation run infinitely (not even increase in epoch count)

### What version are you seeing the problem on?

v1.9

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

cc @carmocca @awaelchli

**Comment by thesofakillers** [comment]
I've also noticed this. Additionally if you check the 'epoch' key in the the 'last' checkpoint, it seems to have reset to '0'. Something is very wrong somewhere when early stopping gets triggered before `min_epochs`. See also #18063

**Comment by thesofakillers** [comment]
Btw, i think this is a duplicate of #15708, which got fixed by #16719

**Comment by jaswanthbjk** [comment]
Thanks for the response @thesofakillers. I will test and close the issue.

**Comment by awaelchli** [comment]
@jaswanthbjk Were you able to check if the fixes in 1.9.5 work for you?
